### PR TITLE
feat(*/*): initial commit

### PR DESCRIPTION
**Pull request by mrogers-5s**

### CODE REVIEW

File: lib/huginn_acumen_order_agent/acumen_order_agent.rb [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+        MD
+
+        def default_options
+            {
+                'endpoint' => 'https://example.com',
+                'site_code' => '',
+                'password' => '',
+                'digital_formats' => [],
+                'attribute_to_property' => {},
```
Comment (jhaltercmu2020) [inline lib/huginn_acumen_order_agent/acumen_order_agent.rb]: You have digital_formats and attribute_to_property default here as default options but they don't seem to be used anywhere

File: lib/huginn_acumen_order_agent/acumen_order_agent.rb [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+                log(":: order codes #{order_codes}")
+            log('------------------------------------------------------------')
+
+            begin
+              invoices = fetch_invoice_data(client, order_codes)
+
+              unless invoices.blank?
+                log('------------------------------------------------------------')
+                    log(":: invoices #{invoices}")
+                log('------------------------------------------------------------')
```
Comment (jhaltercmu2020) [inline lib/huginn_acumen_order_agent/acumen_order_agent.rb]: Do we want to keep these logs? Same with logs on 118, 119, and 120?

File: lib/huginn_acumen_order_agent/concerns/invoice_query_concern.rb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+          'Invoice.Customer_Name' => 'customer',
+          'Invoice.Modified_Date' => 'modified_date',
+        })
+
+        invoice['type'] = '@Order'
+
+        #----------  Parse Tracking Numbers  ----------#
+        invoice['orderDelivery'] = []
+        tracking_numbers = get_field_value(i, 'Invoice.Tracking_Num')
+        tracking_numbers = tracking_numbers.blank? ? [] : tracking_numbers.split(/\s|&#13;/)
```
Comment (jhaltercmu2020) [inline lib/huginn_acumen_order_agent/concerns/invoice_query_concern.rb:40]: What is this `tracking_numbers.split(/\s|&#13;/) doing?

Specifically within the split? Is that encoded characters from acumen?

Comment (mrogers-5s) [reply lib/huginn_acumen_order_agent/concerns/invoice_query_concern.rb:40]: The tracking number field in Acumen is not always a single number. Sometimes it's  a list of tracking numbers.

And Acumen is archaic, so it doesn't store the data as a CSV or an array or any kind of useful format. It stores it with a delimiter of `&#13;` or an HTML encoded `\r` character.

`\s` matches whitespace characters
`|` is basically an `or`
`&#13;` matches that string of characters exactly.

So the the regex is basically saying "look for any white space character or `&#13;`", and the _split_ call it splitting the string on each match it finds. The end result is an array of tracking numbers.

File: lib/huginn_acumen_order_agent/acumen_order_agent.rb (hunk 2 above)
Comment (mrogers-5s) [reply lib/huginn_acumen_order_agent/acumen_order_agent.rb]: No we do not. Fixed

File: lib/huginn_acumen_order_agent/acumen_order_agent.rb (hunk 1 above)
Comment (mrogers-5s) [reply lib/huginn_acumen_order_agent/acumen_order_agent.rb]: Yeah. Those need to go. Fixed

**Review by jhaltercmu2020 — APPROVED** [review]

**Review by Ken-Michalak — APPROVED** [review]

**Review by jhaltercmu2020 — APPROVED** [review]

**Pull request merged**